.github/workflows/tests.yml

```diff
@@ -72,3 +72,23 @@ jobs:
     - name: "Run HLS test suite"
       run: make -j `nproc` -O run
       shell: bash
+
+    - name: "Get the contents of the cycle-diff.log if it exists"
+      id: get-cycles-log
+      if: ${{ hashFiles('build/cycle-diff.log') != '' }}
+      run: |
+        echo "cycles=\"$(cat build/cycle-diff.log | sed -z 's/\n/ | /g' )\"" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: "Create comment if cycles differ"
+      if: ${{ hashFiles('build/cycle-diff.log') != '' }}
+      uses: actions/github-script@v7
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+          github.rest.issues.createComment({
```

Comment:
script: seems to be JavaScript, so maybe it's possible to add code here that reads the file.